### PR TITLE
update docs for kafka lookup extension to specify correct extension loading requirements

### PR DESCRIPTION
**Pull request by georgew5656**
It's not necessary to load both druid-lookups-cached-global and druid-kafka-extraction-namespace to use kafka lookups. In fact if you happen to load both in the wrong order (global lookups before kafka lookups), it actually breaks kafka lookups (see: https://github.com/apache/druid/issues/3538).

**Background**
the druid-kafka-extraction-namespace extension has a dependency on druid-lookups-cached-global, so when loading the extension, both modules (<KafkaExtractionNamespaceModule> and <NamespaceExtractionModule>) are included.

So when druid-kafka-extraction-namespace is included as a extension, both modules are loaded and the features for both extensions are loaded (kafka lookups and global cached lookups)

There is logic in ExtensionsLoader.tryAdd that checks whether a module has already been loaded during initialization and skips it if it already has been loaded.

This is a problem when both druid-kafka-extraction-namespace and druid-lookups-cached-global are specified because they both load NamespaceExtractionModule.

If druid-kafka-extraction-namespace is specified first, both NamespaceExtractionModule and KafkaExtractionNamespaceModule are loaded by the druid-kafka-extraction-namespace classloader, and the druid-lookups-cached-global classloader doesn't load anything since NamespaceExtractionModule was already loaded. This is fine because the features of druid-lookups-cached-global are served through the module NamespaceExtractionModule being loaded in druid-kafka-extraction-namespace. (this is essentially the same behavior as just loading druid-kafka-extraction-namespace, and this is why loading both extensions in this order works)

If druid-lookups-cached-global is specified first, NamespaceExtractionModule is loaded by the druid-lookups-cached-global class loader. The druid-kafka-extraction-namespace classloader will only load KafkaExtractionNamespaceModule because NamespaceExtractionModule has already been loaded. This is a problem because kafka lookups rely on classes bound in NamespaceExtractionModule that it can't access (because NamespaceExtractionModule is only bound in the druid-lookups-cached-global classloader). this is the cause of the linked bug

### Description
IMO the current behavior is a little weird (loading one extension essentially loads two extensions), but to fix this we would have to either move the shared code into druid core or allow first class extension dependencies. For now i think it makes sense to update the documentation to specify that you should only ever load one of the two extensions.

I have tested this behavior in a druid cluster (loading just the kafka extension and trying to create both a kafka lookup and a cachedNamespace lookup)

#### Release note
Update documentation for enabling kafka and globally cached lookups

##### Key changed/added classes in this PR
 * docs/querying/kafka-extraction-namespace.md
 * docs/querying/lookups-cached-global.md

<hr>

This PR has:

- [X] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [X] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [X] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by arunramani** [comment]
Is there a known issues part of the Druid docs? You could also document the error and provide the same explainer there.

File: docs/querying/kafka-extraction-namespace.md [hunk 1]
```diff
@@ -22,7 +22,10 @@ title: "Apache Kafka Lookups"
   ~ under the License.
   -->
 
-To use this Apache Druid extension, [include](../configuration/extensions.md#loading-extensions) `druid-lookups-cached-global` and `druid-kafka-extraction-namespace` in the extensions load list.
+To use this Apache Druid extension, [include](../configuration/extensions.md#loading-extensions) `druid-kafka-extraction-namespace` in the extensions load list.
+
+This extension provides all the functionality of `druid-lookups-cached-global`. You should only load one of the two extensions.
+(load `druid-lookups-cached-global` if you only want global cached lookup functionality, load `druid-kafka-extraction-namespace` if you also want kafka lookup functionality)
```
Comment (techdocsmith) [inline docs/querying/kafka-extraction-namespace.md:28]:
```suggestion
The `druid-kafka-extraction-namespace` extension includes all the functionality of `druid-lookups-cached-global`.
Do not load both extensions, but choose the extension that best suits your needs:
- `druid-kafka-extraction-namespace` if you need Kafka lookup functionality
- `druid-lookups-cached-global` if you only want global cached lookup functionality
```
make admonition to not load both stronger. Capitalize Kafka

File: docs/querying/lookups-cached-global.md [hunk 2]
```diff
@@ -24,6 +24,9 @@ title: "Globally Cached Lookups"
 
 To use this Apache Druid extension, [include](../configuration/extensions.md#loading-extensions) `druid-lookups-cached-global` in the extensions load list.
 
+The `druid-kafka-extraction-namespace` extension provides all the functionality of `druid-lookups-cached-global`. You should only load one of the two extensions.
+(load `druid-lookups-cached-global` if you only want global cached lookup functionality, load `druid-kafka-extraction-namespace` if you also want kafka lookup functionality)
+
```
Comment (techdocsmith) [inline docs/querying/lookups-cached-global.md:29]:
```suggestion
The `druid-kafka-extraction-namespace` extension includes all the functionality of `druid-lookups-cached-global`.
Do not load both extensions, but choose the extension that best suits your needs:
- `druid-lookups-cached-global` if you only want global cached lookup functionality
- `druid-kafka-extraction-namespace` if you need Kafka lookup functionality

```
make admonition to not load both stronger. Capitalize Kafka

**Review by techdocsmith — CHANGES_REQUESTED** [review]
style changes.

**Comment by georgew5656** [comment]
i think i found a better way to fix this bug, so i'm gonna close this docs pr

**Pull request closed without merging**